src/main/java/it/pagopa/dss/SignatureServiceInterface.java

```diff
@@ -0,0 +1,63 @@
+package it.pagopa.dss;
+
+import it.pagopa.dss.exception.SignatureServiceException;
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+
+public interface SignatureServiceInterface {
+  /**
+   * Add a PAdES structure to an input PDF and save it in an outputStream
+   * @param originalPdf pdf file to which to add the PAdES structure
+   * @param outputStream outputstream on which to write the PDF file with the added PAdES structure
+   * @return void
+   * @throws IOException
+   */
+  void generatePadesFile(File originalPdf, OutputStream outputStream) throws IOException;
+
+  /**
+   * Add a PAdES structure to an input PDF and save it in an outputStream
+   * @param originalPdf pdf file to which to add the PAdES structure
+   * @param outputStream outputstream on which to write the PDF file with the added PAdES structure
+   * @param signatureFieldId id of the signature field present on the PDF file
+   * @return void
+   * @throws IOException
+   */
+  void generatePadesFile(
+    File originalPdf,
+    OutputStream outputStream,
+    String signatureFieldId
```

Comment:
what if we have multiple signature fields?

Reply:
No problem, you have to indicate the signatureFieldId. If the signature already exists on that signatureFieldId an exception is thrown!